pkg/kubelet/kubelet_pods.go

```diff
@@ -1245,6 +1255,36 @@ func (kl *Kubelet) HandlePodCleanups(ctx context.Context) error {
 	return nil
 }
 
+// Filters terminal pods with deletion timestamp which are not runtime pods
```

Comment:
Would prefer a slightly longer godoc that covers the full logic.

Reply:
Added the godoc.